lab4/src/main/java/lab4/main/Parser.java

```diff
@@ -0,0 +1,33 @@
+package lab4.main;
+
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+import java.io.File;
+
+public class Parser {
+    private Scanner s;
+    private final String pattern;
+
+    public Parser(String file, String pattern) {
+        this.pattern = pattern;
+        try {
+            this.s = new Scanner(new File(file));
+            findPattern();
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found: " + file);
+        }
+    }
+
+    public void findPattern() {
+        int i = 0;
```

Comment:
i should be = 1 since the first line is 1

Reply:
This has been fixed!

Reply:
Good!